### ISSUE TITLE
Overkill Restarting On File Opening

### ISSUE THREAD

**Issue by atom0s** [remy/nodemon#1208]
- `nodemon -v`: 1.14.7
- `node -v`: 8.9.4
- Operating system/terminal environment: Windows 10 (and Windows 8.1)
- Command you ran:

nodemon

### Expected behaviour

Restarts triggered when a source file is altered.

### Actual behaviour

Restarts are triggering just opening the files in an editor such as VSCode.

### Steps to reproduce

1. Install node.
2. Install nodemon.
3. Run nodemon in a source directory.
4. Open the source in VSCode and just click in the file browser to open the file in the editor.

Soon as the file is opened, Nodemon will trigger a restart/refresh.

Seems a bit overkill to be triggering just when the file is opened in an editor such as VSCode. If I run with legacy mode (-L) this stops happening but then I have the issue of every file being polled excessively as the docs state. 

Everything is working as it should otherwise, but the retriggering while using VSCode seems a bit excessive. I can't find any way to turn this off either or prevent it from happening so that it goes back to only updating when a file is saved.

**Comment by atom0s** [comment]
Running with verbose mode on, the trigger is seen as:

```
[nodemon] files triggering change check: some\file\path\data\1000.json
[nodemon] matched rule: **\*.*
[nodemon] changes after filters (before/after): 1/1
[nodemon] restarting due to changes...
[nodemon] some\file\path\data\1000.json
```

Just clicking on the 1000.json file to open it in VSCode editor causes the trigger and restart.

**Comment by remy** [comment]
What, in Windows, are you using to run nodemon? Is it the command prompt, powershell, git bash, or something else? Also, I suspect this is because somehow the OS is triggering a notify of file change - rather than nodemon doing anything special…

**Comment by atom0s** [comment]
I am using Powershell.

**Comment by remy** [comment]
@atom0s are you able to replicate in a non-powershell? (just to rule out vs code or something more general)

**Comment by atom0s** [comment]
Yes, it happens with a normal command prompt as well. Just tested. 
Updated to latest nodemon (1.14.8) and is still happening as well.

**Comment by remy** [comment]
Can you re-test with nodemon@1.14.10? There's been a few changes that might affect your testing (I couldn't replicate with powershell on my own windows machine).

**Comment by SpencerKaiser** [comment]
I'm having a similar issue... when using atom if I have multiple files open (not just being previewed) and I switch between them (e.g., switch tabs), the file I switch to is triggering a restart. No saves are taking place and nothing should be modifying those files. 

Just updated and verified the issue again with nodemon@1.14.11, please let me know if any other info would be helpful!

**Comment by remy** [comment]
@SpencerKaiser can you include a screenshot of the nodemon output with `--verbose` flag when it's restarting. I'm suspecting the editor touching some files, but I can't replicate this at all so it's hard to know.

**Comment by SpencerKaiser** [comment]
Maybe git is modifying files or my editor is modifying them for me on git's behalf because it seems like the restart only triggers for files with unstaged changes

**Comment by remy** [comment]
@atom0s I'm not familiar with procmon, so I'm not sure why (or what) is walking through your current working path, but can you include the output of `nodemon -V` so I can look for any config that might help diagnose. 

Like I said, I can't replicate at the moment, so I'm unsure of where the root of the issue is.

**Comment by alexisdiel** [comment]
When express requests the file to render view, it detects as a change and reloads 

> [14:41:35] [nodemon] files triggering change check: views\card.pug
> [14:41:35] [nodemon] matched rule: **\*.*
> [14:41:35] [nodemon] changes after filters (before/after): 1/0

**Comment by remy** [comment]
@alexisdiel I'm guessing you have a nodemon config in place, only because you're currently watching everything with **.

**Comment by alexisdiel** [comment]
Sure @remy  
I'm using it along with browserSync and fixed by not watching some folders, configuring the browserSync to watch these folders.

**Comment by remy** [comment]
@atom0s still waiting on a reply. Any chance you're able to follow up? Or I can close and re-open if others are seeing this issue (in a way that it's replicable).

**Comment by mikemaccana** [comment]
Same as @alexisdiel - express opens a `hbs` file and it triggers an update. I should be able to produce a minimal test case to help replicate if needed/wanted.

**Comment by arasmussen** [comment]
Not sure what the issue is, but reinstalling fixed for me.

```sh
npm uninstall -g nodemon
npm install -g nodemon
```

**Comment by akwilson** [comment]
I am also affected by this issue, but only since the 1803 Window Update in April.

Node; v8.11.1
Nodemon: v1.17.3
Windows 10 build 1803
Ubuntu on WSL v16.04.4

Something is triggering a restart when opening a file, but not consistently.  Restarts are also triggered almost at random for no obvious reason.  I checked the timestamps on the files on the project -- they are not updating.

Actually, I don't think this is a problem with nodemon specifically.  I'm adding my comment here because this thread is the closest thing I have found to anyone else suffering from this problem!  I also have a React project that uses webpack which also suffers from similar annoying restarts.

**Comment by mikemaccana** [comment]
This is with no editors open, just running gulp-nodemon, and with `verbose` enabled

    [13:07:10] [nodemon] 1.17.4
    [13:07:10] [nodemon] to restart at any time, enter `rs`
    [13:07:10] [nodemon] watching: C:\Users\mike\Documents\certsimple\routes/**/* C:\Users\mike\Documents\certsimple\views/**/*
    [13:07:10] [nodemon] watching extensions: js
    [13:07:10] [nodemon] starting `node bin/www debug`
    [13:07:10] [nodemon] forking
    [13:07:10] [nodemon] child pid: 16500
      Finished JS bundle index.
    [13:07:11] [nodemon] watching 15 files
      Finished JS bundle order-management.
    [13:07:11] [nodemon]
    [13:07:11] [nodemon] files triggering change check: routes\index.js
    [13:07:11] [nodemon] matched rule: C:\Users\mike\Documents\certsimple\routes/**/*
    [13:07:11] [nodemon] changes after filters (before/after): 1/1
    [13:07:11] [nodemon] restarting due to changes...
    [13:07:11] [nodemon] Routes\index.js
    [13:07:11] [nodemon] starting `node bin/www debug`
    [13:07:11] [nodemon] forking
    [13:07:11] [nodemon] child pid: 14020

Some interestng points:

 - It's the routes file triggering the updates, but it's definitely not changing. 
 - The NTFS `LastAccessTime` is changing:
 
    $ date
    Tuesday, June 5, 2018 1:10:24 PM


    $ ls .\routes\index.js | select *

    CreationTime      : 13-Nov-17 1:42:07 PM
    CreationTimeUtc   : 13-Nov-17 1:42:07 PM
    LastAccessTime    : 05-Jun-18 1:07:12 PM
    LastAccessTimeUtc : 05-Jun-18 12:07:12 PM
    LastWriteTime     : 31-May-18 9:19:32 AM
    LastWriteTimeUtc  : 31-May-18 8:19:32 AM

So my general feeling is that `LastAccessTime` modifications are triggering restarts

**Comment by svogal** [comment]
> So my general feeling is that `LastAccessTime` modifications are triggering restarts

Yes!
Running this command as administrator did the trick:
`fsutil behavior set disablelastaccess 1`

**Comment by mikemaccana** [comment]
Thanks for the confirmation @svogal. With that in mind I've checked this out looking at fixing it. Check out `watch.js:52` - watcher sends it's results to `filterAndRestart()` but `filterAndRestart()` only gets a list of files, not what aspect of them changed - I was hoping to filter out `LastAccessTime` changes but it looks like that needs to be done in `chokidar`.

**Comment by mikemaccana** [comment]
@remy this isn't stale, the issue (to sum up the discussion above) is fixed in https://github.com/paulmillr/chokidar/pull/762 and once chokidar do a release, and nodemon uses it, the problem will go away.

**Comment by mikemaccana** [comment]
I've raised a separate issue since thus one is closed - all `nodemon` needs to do is update `chokidar`.
